### ISSUE TITLE
Backstage Membership Google Account Information

### ISSUE THREAD

**Issue by hjbwynne** [backstagetechnicalservices/website#53]
With the move to Google Drive for our event folders, it would be good if the website could capture a Google Account enabled email address in addition to a member's main Backstage email address. A list could then be created to make file sharing with the membership easier

**Comment by bnjns** [comment]
I'm not sure this would actually bring much (if any) benefit. The sharing needs to be done directly from Drive anyway, so there wouldn't be a way to use this list. Plus, you end up collecting everyone's Google accounts when you do events, and it'll be so much quicker to just use that than have to search through the website.

I'm also very hesitant to collect data that isn't necessary for the direct functionality of the website.

**Comment by ACascarino** [comment]
If you're really intent on making all event folders open access (which, by the way, I'm really hesitant about for a number of reasons but hey, it's your prerogative) then perhaps we could embed an members-only invite link on the website that automatically shares the folder. That way, we don't collect any data ourselves, whilst still providing some standard method by which to add members.

Alternatively, we could have the website send a query to the Google API adding people's xxxxx@bath.ac.uk email addresses to the folder, which will automatically send them invite links when they are added to the website.

**Comment by bnjns** [comment]
We 100% cannot make the Google Drive folders open-access - we often have copyright-restricted material on shows and opening the GDrive folders would be a very bad idea. 

Good idea on an invite-link, definitely the best way to do it without needing to collect more data :)

**Comment by ACascarino** [comment]
Sorry, clarification there - I meant "open access" to mean "adding all members to all event folders" rather than the current "only add event crew to a particular event's folder"; certainly wasn't implying that Harry's trying to open it to the whole wide world!

**Comment by bnjns** [comment]
Ah, got you. 

I don't see any need to be strict about keeping them "crew-only", but of course if that's possible it would be better. IIRC, the API is a bit of a pain to work with but I'm sure I can look into a good way to do it when I get around to it.

**Comment by hjbwynne** [comment]
We must remember that most people’s Bath email address is not Google Drive enabled, hence the collection of alternative emails.

You cannot edit google drive folders without being signed into Google Drive and I’m not sure where the security on links stands.

**Comment by ACascarino** [comment]
Correct, nobody's Bath email address is a Google account. However, if you try to add a non-Google account to a shared GDrive folder, Google will send a link to that folder to that email address. They will then sign in with their personal Google account, or create a Google account if they do not have one, and be required to Request Access to the folder with their personal account, Once this is approved, they will have access to the folder. The same process is followed if you supply a link to the folder yourself, for example by linking to a folder from the website.

This disconnect between the user's Bath address and their Google account is irritating, but personally I'm uncomfortable with us collecting users' Google accounts in a database on the website for security reasons. The workaround would be to have the folder truly "open access", which as Ben points out is infeasible - at some point there has to be some kind of access control.

**Comment by bnjns** [comment]
I think Angel's idea is a good solution to this - will do what you want without needing to collect more data.

**Comment by ACascarino** [comment]
Copying part of a conversation with Jamie below with a suggestion for a proposed system:

"We can solve the issue that - as you note - anyone with access can see who else has access, as well as create a really nice administrative platform for ourselves.
Instead of adding all the members to the folder directly, we add them all to a closed Google Group and give the group edit access to the folder. That means the owner of the group is the only person who can see the members of the group - everyone else will just see that the group has edit access, but won't see who else is in the group and therefore everyone else's person details. 
This also means that a person can opt out of all BTS folder access with a single button - leaving the group removes all their permissions from everything in the folder. By extension, it also means that we can very tightly regulate who has access to the folder - the group members are the only people who have access to anything, and removing a member removes them from everywhere simultaneously.

The only time that anyone else would see anyone else would be if an edit was made to a document, at which point they see the name but not the email address."

This would be implemented here as putting an invite link to the group on the website.
@hjbwynne Up to you how you implement and I've already said to Jamie it's between you two, just thought I'd write this down so it didn't get lost somewhere

**Comment by bnjns** [comment]
This is getting very off-topic, but I can't help but note that is there really a need to do any of this? I don't see why being able to see who's a member of a folder is a problem (after all, you know who's on crew ...) and why is there a need to hide the folder from non-crew? If anything, surely we want it more open so people can use the information for reference in other events? That's what used to happen when we used the Resources drive.

**Comment by hjbwynne** [comment]
Thanks Angel.

I agree it’s best to regulate access as best as we can. The form also collects the graduation year. I did thing about creating a group before but dismissed it for the effort. 
Have to weigh up the pros and cons of course but but the form collects the expected graduation year for each member (or whether they are an associate) so we can group our members accordingly and work out when to receive access.

I will discuss the options with Jamie when I have more names but using groups could also make it possible to create private TEM folders within the public folder system with ease.

Ben, whilst yes it seems that we are creating more admin work for ourselves, it is also worth noting that the SU will want to make sure we are keeping the data as private as possible especially with the (mostly unnecessary) worrying about GDPR that is going in within large organisations at the moment.

**Comment by bnjns** [comment]
My issues have nothing to do with extra admin, but the fact that you seem to be wanting to hide information from certain members. Yes, at the moment only event crew have access to their folders but it has always been the plan of the PM to download each event folder and add them to the resources drive once the event is done - it's just no one has ever done it. 

Everyone in the society should be treated equally and be given access to the same information - especially when it comes to event folders as being able to see what previous events have done is extremely useful when planning an event. Also, associates are also members and shouldn't have access to event folders removed - many associates provide invaluable experience and input to many events throughout the year.  

This has also got nothing to do with GDPR - that's not a go to reason for restricting access to something.

**Comment by ACascarino** [comment]
I think there's been a lot of miscommunication here so at the risk of wading in I'll try and clarify things as I see them.

 - The intent from Harry is to have an events folder accessible to everyone.
 - This was to be implemented by gathering Google account details from everyone and adding them as "shared" to the folder individually.
 - I pointed out that us keeping a repository of members' _personal_ information (Google accounts) is not great, and we should look to implement a system that allows people to see as little _personal information_ about each other as possible.
 - The proposed Groups solution does this, whilst enabling everyone to be able to contribute to a shared folder.

I think we're actually all agreeing at this point on the core aims:
 - More access
 - Less personal information on display
 - Tighter access control
 - Not constrained to users with BUCS accounts.

**Comment by bnjns** [comment]
Thanks for clarifying. This is getting very off-topic / confused so I'm going to tidy the comments up later.

In summary, we will be integrating the event Google Drive folders to the event page, and will try to automatically add access to said folders when people are added to the crew. 

I ask that any replies are directly related to the website to avoid any more confusion.
